### ISSUE TITLE
Support setting GTK+ dark theme variant

### ISSUE THREAD

**Issue by p-e-w** [electron/electron#352]
GTK+ supports a [dark theme variant](https://developer.gnome.org/gtk3/3.0/GtkSettings.html#GtkSettings--gtk-application-prefer-dark-theme) which, mainly on Linux-based systems, allows for dark window decorations that blend much better with a dark chrome such as the one Atom uses by default:
### Before

![Before](https://raw.githubusercontent.com/p-e-w/GTKDarkThemeVariantSetter/images/before.png)
### After

![Before](https://raw.githubusercontent.com/p-e-w/GTKDarkThemeVariantSetter/images/after.png)

(Screenshots are from my Sublime Text plugin https://github.com/p-e-w/GTKDarkThemeVariantSetter which enables the dark theme variant for Sublime).

<br>

I believe it would be a very nice addition if atom-shell supported a mechanism that allows the application running inside of it to choose this variant rather than the standard one. The problem of dark-themed editors looking ugly on GTK is a serious one, prompting such efforts as @nikolowry's https://github.com/nikolowry/atom-csd (which of course offers much more than just a dark theme for the window decorations).

<br>

Basically, there are two ways to achieve this:
1. Make atom-shell a `GtkApplication` and set the `gtk-application-prefer-dark-theme` property, **or**
2. Set the `_GTK_THEME_VARIANT` property to `dark` on the `GtkWindow`

I don't yet fully understand the way atom-shell exposes functionality to node but if you are OK with this idea and willing to give me a few code pointers I'd be glad to implement it.
